model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/remotestore/RemoteCacheInvoker.java

```diff
@@ -146,6 +169,18 @@ private <K, V extends SessionEntity> void replace(TopologyInfo topology, RemoteC
 
             VersionedValue<SessionEntityWrapper<V>> versioned = remoteCache.getWithMetadata(key);
             if (versioned == null) {
+                if (Profile.isFeatureEnabled(Profile.Feature.PERSISTENT_USER_SESSIONS) &&
+                    (remoteCache.getName().equals(USER_SESSION_CACHE_NAME)
```

Comment:
You don't have to check the cache name and invoke remove in all the scenarios.

Reply:
IMHO I need to check the cache name, as this is also used for login failures.